src/main/haskell/kore/test/Test/Kore/Step/Pattern.hs

```diff
@@ -0,0 +1,137 @@
+module Test.Kore.Step.Pattern where
+
+import Test.Tasty
+       ( TestTree )
+import Test.Tasty.HUnit
+       ( testCase )
+
+import           Data.Map.Strict
+                 ( Map )
+import qualified Data.Map.Strict as Map
+
+import Control.Monad.Counter
+import Kore.AST.Common
+import Kore.AST.Valid
+import Kore.Step.Pattern
+import Kore.Variables.Fresh
+
+import           Test.Kore.Comparators ()
+import qualified Test.Kore.Step.MockSymbols as Mock
+import           Test.Tasty.HUnit.Extensions
+
+test_substitute :: [TestTree]
+test_substitute =
+    [ testCase "Variable: Target"
+        (assertEqualWithExplanation ""
+            (mkVar Mock.z)
+            (evalCounter $ substitute' (mkVar Mock.x))
```

Comment:
I'm not a fan of tests that need scrolling in order to be understood. I know that sometimes it's the best option, but here how about substituting `substitute'` with `substituteXWithZ` or inlining it?